homeassistant/components/camera/onvif.py

```diff
@@ -114,9 +119,17 @@ def __init__(self, hass, config):
                 config.get(CONF_USERNAME), config.get(CONF_PASSWORD)
             )
             media_service = camera.create_media_service()
-            stream_uri = media_service.GetStreamUri(
-                {'StreamSetup': {'Stream': 'RTP-Unicast', 'Transport': 'RTSP'}}
-                )
+            self._profiles = media_service.GetProfiles()
```

Comment:
You are not allowed to do I/O inside the constructor as the constructor is called from an async context.

Reply:
You can change the context to be sync by dropping the `@asyncio.coroutine` decorator and make it `def setup_platform(hass, config, add_devices, discovery_info=None)`